src/MagnumPlugins/AssimpImporter/AssimpImporter.cpp

```diff
@@ -0,0 +1,526 @@
+        if(err == nullptr) {
+            Error() << "Trade::AssimpImporter::image2D(): Embedded texture path did not contain a valid integer string.";
+            return std::nullopt;
+        }
+        const aiTexture* texture = _scene->mTextures[index];
+
+        if(texture->mHeight == 0) {
+            /* Compressed image data */
+            auto textureData = Containers::ArrayView<const char>(reinterpret_cast<const char*>(texture->pcData), texture->mWidth);
+
+            std::string importerName;
+            if(texture->CheckFormat("dds")) {
+                importerName = "DdsImporter";
+            } else if(texture->CheckFormat("jpg")) {
+                importerName = "JpegImporter";
+            } else if(texture->CheckFormat("pcx")) {
+                importerName = "PxcImporter";
+            } else if(texture->CheckFormat("png") || std::strncmp(textureData.suffix(1).data(), "PNG", 3) == 0) {
+                importerName = "PngImporter";
+            } else {
+                Error() << "Trade::AssimpImporter::image2D(): Could not detect filetype of embedded data.";
+                return std::nullopt;
+            }
+
+            std::unique_ptr<Trade::AbstractImporter> importer =
+                static_cast<PluginManager::Manager<Trade::AbstractImporter>&>(*manager()).loadAndInstantiate(importerName);
+            if(!importer) {
+                Error() << "Trade::AssimpImporter::image2D(): Could not find importer for embedded data.";
+                return std::nullopt;
+            }
+            importer->openData(textureData);
+            return importer->image2D(0);
+        } else {
+            /* Uncompressed image data */
+            const Vector2i dimensions{Int(texture->mHeight), Int(texture->mWidth)};
+            size_t size = dimensions.product()*4;
+            Containers::Array<char> data{Containers::NoInit, size};
+            std::memcpy(texture->pcData, data.data(), size);
+
+            return ImageData2D(PixelFormat::RGBA, PixelType::UnsignedByte, dimensions, std::move(data), texture);
+        }
+    } else {
+        /* Load external texture */
+        AnyImageImporter importer{static_cast<PluginManager::Manager<AbstractImporter>&>(*manager())};
+        if(_filePath) {
+            importer.openFile(Utility::Directory::join(*_filePath, path));
+        } else {
+            importer.openFile(path);
+        }
+        return importer.image2D(0);
+    }
+}
+
+const void* AssimpImporter::doImporterState() const {
+    return _scene;
+}
+
+void AssimpImporter::createNodeIndices() {
+    if(!_scene) {
+        /* Scene was not imported correctly by assimp */
+        return;
+    }
+    if(!_nodes.empty()) {
+        /* Already done previously */
+        return;
+    }
+
+    /* reserve memory for the amount of nodes we know will at least be present */
+    aiNode* root = _scene->mRootNode;
+    if(!root) {
+        return; /* Nothing to do */
+    }
```

Comment:
Coding style: this could be all one-liners with `return` on the same line, I don't enforce braces everywhere.